indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamSequenceNumbers.java

```diff
@@ -53,6 +53,15 @@ SeekableStreamSequenceNumbers<PartitionIdType, SequenceOffsetType> plus(
       SeekableStreamSequenceNumbers<PartitionIdType, SequenceOffsetType> other
   );
 
+  /**
+   * Return True if this is greater than other instance sequence.
+   *
+   * @see DataSourceMetadata#isGreater
+   */
+  boolean isGreater(
+          SeekableStreamSequenceNumbers<PartitionIdType, SequenceOffsetType> other
+  );
```

Comment:
how about implementing the `Comparable` interface for this instead of rolling our own?

Reply:
@xvrl, I have implemented the Comparable interface. I am thinking of removing the existing match function as well. , we can do all three greaterThan, LessThan and equal in the same compareTo function. Is it good idea to do that?